# What is Symbolic Computation?

Symbolic computation deals with the computation of mathematical objects symbolically. This means that the mathematical objects are represented exactly, not approximately, and mathematical expressions with unevaluated variables are left in symbolic form.

Let's take an example. Say we wanted to use the build-in Python functions to compute square roots. We might do something like this

In [14]:
import math
math.sqrt(9)

3.0

9 is a perfect square, so we got the exact answer, 3. But suppose we computed the sqaure root of a number that isn't a perfect square.

In [15]:
math.sqrt(8)

2.8284271247461903

Here we got an aprroximate result. 2.82842712475 is not the exact square root of 8 (indeed, the actual square root of 8 cannot be represented by a finite decimal, since it is an irrational number). If all we cared about was the decimal form of the square root of 8, we would be done.

But suppose we want to go further. Recall that ![](img/1.png) We would have a hard time deducing this form the above result. This is where symbolic computation comes in. With a symbolic computation system like SymPy, square roots of numbers that are not perfect squares are left unevaluated by default

In [17]:
import sympy
sympy.sqrt(3)

sqrt(3)

In [18]:
sympy.sqrt(8)

2*sqrt(2)